### ISSUE TITLE
[Nested Tabstop] nested tabstop still be there even if outer is deleted

### ISSUE THREAD

**Issue by hiberabyss** [L3MON4D3/LuaSnip#565]
```
snippet cc "Blade cc_ item" b
  cc_${1:library}(
    name = '${2:`expand('#:t:r')`}',
    hdrs = [
      '$2.h',
    ],
    ${3:srcs = [
      '$2.cc',
    ],}
    ${4:deps = [
      $5
    ],}
  )
```

For the snip, even if `$3` is deleted, the `$2.cc` still be there. After changing $2, things messed up:

![image](https://user-images.githubusercontent.com/2898670/187672442-4e4506d7-f44f-4d74-8da4-ef8eda0eff82.png)

**Comment by L3MON4D3** [comment]
Yeah, this issue can't happen with the default `choiceNode`-approach for nesting placeholders, which is one reason for having this in the wiki and not in luasnip.

If I have time I'll look into how we can delete the nested nodes with the `parser_nested_assembler` from the wiki.
My main problem is that I'm not sure how to reliably detect that the default-content (eg. the SELECTed region) is cleared.

Maybe via an autocommand after jumping into it.. I'll have to look into that.

**Comment by hiberabyss** [comment]
Looking forward to it!
